lib/modules/watcher/index.js

```diff
@@ -150,6 +162,27 @@ class Watch {
     );
   }
 
+  watchPipelineConfig(embarkConfig, callback) {
+    let dappPipelineConfigPath;
+
+    let filesToWatch = [
+      fs.dappPath('', DAPP_WEBPACK_CONFIG_FILE),
+      fs.dappPath('', DAPP_BABEL_LOADER_OVERRIDES_CONFIG_FILE)
+    ]
+
+    if (typeof embarkConfig.config === 'object' && embarkConfig.config.pipeline) {
+      filesToWatch.push(embarkConfig.config.pipeline);
+    } else if (typeof embarkConfig.config === 'string') {
+      filesToWatch.push(fs.dappPath(embarkConfig.config, DAPP_PIPELINE_CONFIG_FILE));
+    }
```

Comment:
Here's another spot where we need to be careful. `config` could be an object ("simple app") but `pipeline` may not be defined because it might be an existing app that doesn't know this config yet and we want to stay backwards compatible.

If it doesn't have a `pipeline` property, we still have to check whether it's a string because it could still be an object, but `fs.dappPath()` will blow up if its parameters aren't strings.